### PR TITLE
Add Entry.data_for(type, slug)

### LINKED ISSUE #23: Add Entry.data_for(type, slug)
I'll open an issue for implementing this, it's certainly an API I'd like to have in this extension

_Originally posted by @elia in https://github.com/_render_node/MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDQyNDM4ODk3NA==/comments/review_comment_

### PR DESCRIPTION
**Pull request by peterberkenbosch**
Introduces the `Entry.data_for` method. It takes a `EntryType` name and the `Entry` slug. 

- [x] Update README
- [x] Refactor to use this pattern in the controllers
- [x] Refactor to return an entry instance and not the data directly

Fixes #23

### CODE REVIEW

File: spec/models/solidus_content/entry_spec.rb [hunk 1]
```diff
@@ -1,52 +1,58 @@
...
 RSpec.describe SolidusContent::Entry do
   let(:entry) { build(:entry) }
 
-  it 'has a valid factory' do
+  it "has a valid factory" do
     expect(entry).to be_valid
   end
 
-  describe "#data_for" do
+  describe ".data_for" do
```
Comment (elia) [inline spec/models/solidus_content/entry_spec.rb:31]: Thanks! 😅

File: spec/models/solidus_content/entry_spec.rb [hunk 2]
```diff
@@ -13,10 +13,28 @@
     let(:post) { create(:entry_type, name: :post, provider_name: :raw) }
     let!(:post_one) { create(:entry, slug: 'one', options: {title: "first post"}, entry_type: post)}
 
-    it "returns data for type and slug" do
-      expect(
-        SolidusContent::Entry.data_for(:post, 'one')
-      ).to eq({title: "first post"})
+    context "with using correct type and slug" do
```
Comment (elia) [inline spec/models/solidus_content/entry_spec.rb:32]: 👍 can you squash [this commit](https://github.com/nebulab/solidus_content/pull/29/commits/094cd02989ca3ac18d0b59e1f360c0dc10badea9) to the previous one? This way we'll have the change along with its specs together, and also fix the comments that slipped into the commit message 😜

File: app/models/solidus_content/entry_type.rb [hunk 3]
```diff
@@ -2,6 +2,8 @@ class SolidusContent::EntryType < ActiveRecord::Base
   has_many :entries
   after_initialize { self.options ||= {} }
 
+  scope :by_type_name, ->(type_name) { find_by!(name: type_name) }
```
Comment (elia) [inline app/models/solidus_content/entry_type.rb]: I think `by_name` would be as clear, plus shorter, any reason we should avoid it?

File: app/models/solidus_content/entry.rb [hunk 4]
```diff
@@ -6,8 +6,12 @@ class SolidusContent::Entry < ActiveRecord::Base
...
   scope :by_slug, ->(slug) { find_by!(slug: slug) }
 
-  def self.data_for(entry_type, slug)
-    SolidusContent::EntryType.find_by!(name: entry_type).entries.by_slug(slug).data
+  def self.data_for(entry_type_name, slug)
+    get(entry_type_name, slug).data
+  end
+
+  def self.get(entry_type_name, slug)
+    SolidusContent::EntryType.by_type_name(entry_type_name).entries.by_slug(slug)
```
Comment (elia) [inline app/models/solidus_content/entry.rb]: This is great! At this point we could even extract:

```rb
def self.by_type(type)
  unless type.is_a? SolidusContent::EntryType
    type = SolidusContent::EntryType.by_name(type)
  end
  where(entry_type: type)
end
```

and use `SolidusContent::Entry.by_type(:foo).by_slug(:bar)` everywhere, wdyt?

**Review by elia — CHANGES_REQUESTED** [review]
Got some suggestions, please let me know what you think 👍 
Also some of the commits have dangling comments at the end of the description, I guess generated by the editor.

File: app/models/solidus_content/entry.rb (hunk 4 above)
Comment (peterberkenbosch) [reply app/models/solidus_content/entry.rb]: I like that! Refactoring that now.

**Review by elia — APPROVED** [review]

**Pull request merged**